readthedocs/doc_builder/backends/linguist.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import absolute_import
+import json
+import logging
+import os
+import re
+
+from readthedocs.projects.utils import safe_write
+from ..base import BaseBuilder
+from ..environments import BuildCommand
+
+log = logging.getLogger(__name__)
+
+
+class LinguistBuilder(BaseBuilder):
+
+    """Linguist builder which generates a programming language breakdown."""
+
+    type = 'linguist'
+    builder = 'build'
+    build_dir = '_build/linguist'
+
+    LINGUIST_COMMAND_LOCAL = 'linguist'
+    LINGUIST_COMMAND_DOCKER = 'github-linguist'
+
+    LINGUIST_OUTPUT_REGEX = re.compile(
+        r'^(?P<percentage>\d+\.\d+)\%\s+(?P<language>.+)$',
+        re.MULTILINE,
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(LinguistBuilder, self).__init__(*args, **kwargs)
+        self.root_path = self.version.project.checkout_path(self.version.slug)
+        self.old_artifact_path = os.path.join(self.root_path, self.build_dir)
+        self.output_dir = self.old_artifact_path
+
+        # Outputs
+        self.linguist_txt_path = os.path.join(self.output_dir, 'linguist.txt')
+        self.linguist_json_path = os.path.join(self.output_dir, 'linguist.json')
+        self.linguist_error_path = os.path.join(self.output_dir, 'linguist-error.txt')
+
+    def _parse_linguist_output(self, output):
+        data = self.LINGUIST_OUTPUT_REGEX.findall(output)
+        return [(float(perc), lang) for perc, lang in data]
+
+    def _save_linguist_output(self, output):
+        if not output:
+            output = ''
+
+        safe_write(self.linguist_txt_path, output)
+
+        # Parse the language breakdown for easier consumption
+        safe_write(self.linguist_json_path, json.dumps(
+            self._parse_linguist_output(output)
+        ))
+
+    def _save_linguist_error(self, output):
+        if not output:
+            output = ''
+
+        safe_write(self.linguist_error_path, output)
+
+    def build(self):
+        if self.build_env.command_class == BuildCommand:
+            command = self.LINGUIST_COMMAND_LOCAL
+        else:
+            command = self.LINGUIST_COMMAND_DOCKER
```

Comment:
This still feels silly. Can we really not get the same command name on both? Can we just alias it or something in docker?

Reply:
That is probably a better approach.